app/src/main/java/io/gnosis/kouban/ui/balances/BalancesWidgetPrefs.kt

```diff
@@ -17,18 +17,20 @@ class BalancesWidgetPrefs(context: Context) {
 
     fun saveTokenForWidget(token: Solidity.Address, appWidgetId: Int) {
         prefs.edit {
-            putString(KEY_WIDGET_TOKEN, "${appWidgetId}.${token.asEthereumAddressString()}")
+            putString("$KEY_WIDGET_TOKEN_$appWidgetId", token.asEthereumAddressString())
+            putInt("$KEY_WIDGET_ID_$appWidgetId", appWidgetId)
```

Comment:
This is not really required, right?

Reply:
We need to save selected token and for what widget it was selected. Otherwise, we won't be able to update it correctly if we have more than 1 widget added.

Reply:
so the selected token is safe by using a key that is based on the widget id. Therefore you do not need so safe the weidget id separately, right?

This value is also never read, therefore I assumed it is not required ;)

Reply:
🤦‍♂ yes, sure. Later if we decide to support multiple safes we would need to save safe address as well